common/djangoapps/request_cache/middleware.py

```diff
@@ -58,3 +58,50 @@ def process_exception(self, request, exception):  # pylint: disable=unused-argum
         """
         self.clear_request_cache()
         return None
+
+
+def request_cached(wrapped):
+    """
+    Wraps a function and automatically handles caching its return value, as well as
+    returning that cached value for subsequent calls to the same function, with the
+    same parameters, within a given request.
+
+    Caveats:
+        - we convert positional arguments to their string equivalent and use them in the cache key, so if something
+          can't be converted to a string, things might blow up
+        - our cache key depends on the positional args (so , holistically, cache efficiency is tied to cardinality)
+        - we use the default request cache, not a named request cache
+        - the cache key is prefixed with the name of the wrapped function, so theoretically, an identically named
+          function elsewhere in the code with the same parameters could overwrite cached values
+    """
```

Comment:
1. Sure.
2. I'm not sure if having to put `@request_cached_function` as a decorator is much clearer?  Maybe something like... `@use_request_cache`?
3. Sure.